PDAL 2.0.1 fails with BUILD_SHARED_LIBS=ON due to error in gtest

I am using `Ubuntu 18.04 LTS (Linux 5.0.0-37-generic #40~18.04.1-Ubuntu)` Using following cmake command:
```bash
cmake -D WITH_LAZPERF=ON \
      -D WITH_LZMA=ON \
      -D BUILD_PLUGIN_PGPOINTCLOUD=ON \
      -D BUILD_PGPOINTCLOUD_TESTS=ON \
      -D BUILD_PLUGIN_SQLITE=ON \
      -D BUILD_SHARED_LIBS=ON \
      -D BUILD_SQLITE_TESTS=ON \
      -D BUILD_PLUGIN_PYTHON=ON ..
```

When building PDAL 2.0.1 with `BUILD_SHARED_LIBS=ON` build failed while compiling gtest because `${gtest_BINARY_DIR}` is not set properly in the file: `PDAL/cmake/gtest.cmake`

Here is another reference of the same issue on osgeo: http://osgeo-org.1560.x6.nabble.com/pdal-PDAL-2-0-1-build-issue-td5415293.html

Error:
```bash
[  8%] Linking CXX shared library /src/libgtest.so
/usr/bin/ld: cannot open output file /src/libgtest.so: No such file or directory
collect2: error: ld returned 1 exit status
vendor/gtest/CMakeFiles/gtest.dir/build.make:94: recipe for target '/src/libgtest.so' failed
make[2]: *** [/src/libgtest.so] Error 1
CMakeFiles/Makefile2:810: recipe for target 'vendor/gtest/CMakeFiles/gtest.dir/all' failed
make[1]: *** [vendor/gtest/CMakeFiles/gtest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  8%] Built target pdal_test_support
[  8%] Linking CXX static library ../../lib/libpdal_arbiter.a
[  8%] Built target pdal_arbiter
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2
```
**Quick fix solution:**
In the file: `PDAL/cmake/gtest.cmake`, I changed the `LIBRARY_OUTPUT_DIRECTORY "${gtest_BINARY_DIR}/src"` to this `    LIBRARY_OUTPUT_DIRECTORY "${ROOT_DIR}/vendor/gtest/src"` and it compiled successfully.

Kindly address this issue

--- a/cmake/gtest.cmake
+++ b/cmake/gtest.cmake
@@ -4,14 +4,3 @@ if (MSVC)
     option(gtest_force_shared_crt "Always use shared Visual C++ run-time DLL" ON)
 endif()
 add_subdirectory(vendor/gtest)
-
-# gtest 1.7.0 has some CMake warnings, so we silence these by setting the
-# following properties.
-set_target_properties(gtest PROPERTIES
-    MACOSX_RPATH ON
-    LIBRARY_OUTPUT_DIRECTORY "${gtest_BINARY_DIR}/src"
-)
-set_target_properties(gtest_main PROPERTIES
-    MACOSX_RPATH ON
-    LIBRARY_OUTPUT_DIRECTORY "${gtest_BINARY_DIR}/src"
-    )

